src/lib/list/selection-list.ts

```diff
@@ -52,8 +52,9 @@ export class MatListOptionBase {}
 export const _MatListOptionMixinBase = mixinDisableRipple(MatListOptionBase);
 
 /** Event emitted by a selection-list whenever the state of an option is changed. */
-export interface MatSelectionListOptionEvent {
+export class MatSelectionListOptionEvent {
   option: MatListOption;
+  selected: boolean;
 }
```

Comment:
I'd keep the consistency between this and others:

/** Change event object emitted by MatSelectionList. */
export class MatSelectionListChange {
  /** The source MatListOption of the event. */
  source: MatListOption;
  /** The new `checked` value of the option. */
  checked: boolean;
}

See [https://github.com/angular/material2/blob/master/src/lib/checkbox/checkbox.ts#L71](checkbox for example).

Reply:
So checked instead of selected? Gotcha.

Reply:
Also adding comments will be useful so it can be shown in docs.

Reply:
By others you mean other checkboxes? Add comments?

Changing it from selected to checked seems like a step in the wrong direction, since it is MatListOption.selected, not MatListOption.checked.

Reply:
You're right. But what about `source` instead of `option`? All `change` events use `source`.

Reply:
Sounds good, I'll change that.